embed/trezorhal/image.c

```diff
@@ -38,11 +38,13 @@ bool image_parse_header(const uint8_t * const data, const uint32_t magic, const
     if (hdr->hdrlen != HEADER_SIZE) return false;
 
     memcpy(&hdr->expiry, data + 8, 4);
+    // TODO: expiry mechanism needs to be ironed out before production or those
+    // devices won't accept expiring bootloaders (due to boardloader write protection).
     if (hdr->expiry != 0) return false;
 
     memcpy(&hdr->codelen, data + 12, 4);
-    if (hdr->hdrlen + hdr->codelen < 4 * 1024) return false;
-    if (hdr->hdrlen + hdr->codelen > maxsize) return false;
+    if (hdr->codelen > (maxsize - hdr->hdrlen)) return false;
```

Comment:
Is there a reason why not keep `((hdr->hdrlen + hdr->codelen) > maxsize)` here to match the following conditions?

Reply:
I was thinking about large values of codelen that could overflow when added to hdrlen and this way seemed a little more intuitive to me, that's all. As long as that first (old) comparison to 4096 uses >= 512, you're bounded/ok. The 4096 seemed a little magical other than that.